tests/unit/nupic/algorithms/sdr_classifier_test.py

```diff
@@ -841,11 +840,25 @@ def testWriteRead(self):
                          learn=True, infer=True)
 
     self.assertEqual(result1.keys(), result2.keys())
+
     for key in result1.keys():
       for i in xrange(len(c1._actualValues)):
         self.assertAlmostEqual(result1[key][i], result2[key][i], 5)
 
 
+  @unittest.skipUnless(
+      capnp, "pycapnp is not installed, skipping serialization test.")
+  def testWriteRead(self):
+    self._doWriteReadChecks(computeBeforeSerializing=True)
+
+  @unittest.skip("Good test to have, but fails on i out of bounds in "
+                 "result1[key][i]. Need to debug.")
```

Comment:
This test reproduces the "deque index out of bounds" in write before this PR's fix. However, with the fix it fails in a different, likely unrelated, way that needs to be debugged, so skipping for now:
```
======================================================================
ERROR: testWriteReadNoComputeBeforeSerializing (__main__.SDRClassifierTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "sdr_classifier_test.py", line 859, in testWriteReadNoComputeBeforeSerializing
    self._doWriteReadChecks(computeBeforeSerializing=False)
  File "sdr_classifier_test.py", line 846, in _doWriteReadChecks
    self.assertAlmostEqual(result1[key][i], result2[key][i], 5)
IndexError: list index out of range
```